packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -1762,11 +1790,12 @@ export class SnapController extends BaseController<
         return existingSnap;
       }
 
-      if (this.#featureFlags.dappsCanUpdateSnaps === true) {
-        return await this.updateSnap(origin, snapId, location, versionRange);
-      }
-      throw ethErrors.rpc.invalidParams(
-        `Version mismatch with already installed snap. ${snapId}@${existingSnap.version} doesn't satisfy requested version ${versionRange}.`,
+      return await this.updateSnap(
+        origin,
+        snapId,
+        location,
+        versionRange,
+        false,
```

Comment:
Why do we not want to emit an event here?

Reply:
If an update is called as part of `processRequestedSnap` it may be part of multiple installs/updates, if so, we want to wait with emitting the event until all installations and updates have completed.

Reply:
https://github.com/MetaMask/snaps/blob/f4241feeebe45948c2a3f2cb5ed4330f13007173/packages/snaps-controllers/src/snaps/SnapController.ts#L1738-L1752

Reply:
Can you add a comment to explain this?

Reply:
Done